### PR TITLE
Fix physical device enumeration and add dummy features

### PR DESCRIPTION
**Pull request by msiglreith**
Vulkan loader expects that EnumeratePhysicalDevices will return the same handles on each call. Caching them on startup

### CODE REVIEW

File: libportability-gfx/src/impls.rs [hunk 1]
```diff
@@ -36,14 +36,21 @@ pub extern "C" fn gfxCreateInstance(
...
         env_logger::init();
     }
-    let instance = back::Instance::create("portability", 1);
-    unsafe { *pInstance = Handle::new(instance) };
+
+    let backend = back::Instance::create("portability", 1);
+    let adapters = backend
+        .enumerate_adapters()
+        .into_iter()
+        .map(|adapter| Handle::new(adapter))
```
Comment (kvark) [inline libportability-gfx/src/impls.rs]: nit: `map(Handle::new)`

File: libportability-gfx/src/impls.rs [hunk 2]
```diff
@@ -62,19 +69,17 @@ pub extern "C" fn gfxEnumeratePhysicalDevices(
...
         return VkResult::VK_SUCCESS;
     }
 
     let output = unsafe { slice::from_raw_parts_mut(pPhysicalDevices, *pPhysicalDeviceCount as _) };
-    let count = cmp::min(adapters.len(), output.len());
+    let count = cmp::min(instance.adapters.len(), output.len());
 
-    for (out, adapter) in output.iter_mut().zip(adapters.into_iter()) {
-        *out = DispatchHandle::new(adapter);
+    for (out, adapter) in output.iter_mut().zip(instance.adapters.iter()) {
```
Comment (kvark) [inline libportability-gfx/src/impls.rs]: nit: `copy_from_slice`

File: libportability-gfx/src/impls.rs [hunk 3]
```diff
@@ -256,6 +319,147 @@ pub extern "C" fn gfxGetInstanceProcAddr(
         vkGetPhysicalDeviceSurfaceCapabilitiesKHR, PFN_vkGetPhysicalDeviceSurfaceCapabilitiesKHR => gfxGetPhysicalDeviceSurfaceCapabilitiesKHR,
         vkGetPhysicalDeviceSurfaceFormatsKHR, PFN_vkGetPhysicalDeviceSurfaceFormatsKHR => gfxGetPhysicalDeviceSurfaceFormatsKHR,
         vkGetPhysicalDeviceSurfacePresentModesKHR, PFN_vkGetPhysicalDeviceSurfacePresentModesKHR => gfxGetPhysicalDeviceSurfacePresentModesKHR,
+
+        vkDestroyDevice, PFN_vkDestroyDevice => gfxDestroyDevice,
```
Comment (kvark) [inline libportability-gfx/src/impls.rs]: why are device functions in `gfxGetInstanceProcAddr`?

Comment (msiglreith) [reply libportability-gfx/src/impls.rs]: Afaik, GetInstanceProcAddr has addr of all functions

Comment (msiglreith) [reply libportability-gfx/src/impls.rs]: > Function pointers for all Vulkan commands can be obtained with the command

Comment (kvark) [reply libportability-gfx/src/impls.rs]: Can we share the function gather code with GetDeviceProcAddr?

File: libportability-gfx/src/impls.rs (hunk 1 above)
Comment (msiglreith) [reply libportability-gfx/src/impls.rs]: Done

**Comment by msiglreith** [comment]
Thanks should be ready now

**Pull request merged**